lib/player.rb

```diff
@@ -0,0 +1,13 @@
+class Player
+    attr_reader :name, :deck
+    def initialize(name, deck)
+        @name = name
+        @deck = deck
+    end
+
+    def has_lost?
+        return true if deck.cards.length <= 0
+        false
+    end
```

Comment:
Love the combination of explicit and implicit returns being leveraged in this method, nicely done. I do want to point out one tip, however. Since we know that `deck.cards.length <= 0` will either return `true` or `false`, we can use that to our advantage and refactor the method to look something like this:

```ruby
def has_lost?
  deck.cards.length <= 0
end
```

Once this method evaluates, it's either going to be `true` or `false` and then return it.

The last thing I want to mention is to remember that since we are already inside the class, we can use the instance variables. I know if you check online there's a back and forth as to the preferred way, but Turing pushes using the instance variable over the instance method if you are inside the class, at least in Mod 1.

```ruby
def has_lost?
  @deck.cards.length <= 0
end
```